src/Controllers/Common/ContentController.cs

```diff
@@ -778,9 +778,16 @@ public IActionResult GetAvatarByUserId([FromForm] Guid userId)
 
     [HttpPost]
     [Produces("application/xml")]
-    [Route("ContentWebService.asmx/GetUnselectedPetByTypes")] // used by old SoD (e.g. 1.13)
+    [Route("ContentWebService.asmx/GetUnselectedPetByTypes")] // used by old SoD (e.g. 1.13) and Math Blaster
     [VikingSession(UseLock=false)]
-    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string petTypeIDs, [FromForm] bool active) {
+    public RaisedPetData[]? GetUnselectedPetByTypes(Viking viking, [FromForm] string? apiKey, [FromForm] string? userId, [FromForm] string petTypeIDs, [FromForm] bool active) {
+        if (apiKey != null
```

Comment:
it's a real situation that apiKey is null here ... if not, we rather don't add such checks... and `GetVersion` itself is rather null-resistant ...

Reply:
GetVersion is not null-resistant. A call to `apiKey.EndsWith` is made on line 21.

Reply:
yeah, you are right about `GetVersion`